src/utils/helpers.js

```diff
@@ -113,28 +113,29 @@ const utils = {
 
   addOrderIdRecord(orderId) {
     const orderIdHash = CRC32.str(orderId.toString());
-    let purchaseEventStorage = store.session.get(purchaseEventStorageKey);
+    let purchaseEventStorage = store.local.get(purchaseEventStorageKey);
 
     if (typeof purchaseEventStorage === 'string') {
       purchaseEventStorage = JSON.parse(purchaseEventStorage);
     }
 
     if (purchaseEventStorage) {
       // If the order already exists, do nothing
-      if (purchaseEventStorage[orderIdHash]) {
+      if (purchaseEventStorage.includes(orderIdHash)) {
         return;
       }
 
-      purchaseEventStorage[orderIdHash] = true;
+      if (purchaseEventStorage.length >= 10) {
+        purchaseEventStorage.slice(-9);
```

Comment:
I think this should be `purchaseEventStorage = purchaseEventStorage.slice(-9)` 🤔 

Probably not necessary, but should we add a test around this? 😛